app/controllers/MediaController.java

```diff
@@ -231,7 +249,13 @@ public Result upload(Long folderId) {
             } catch (Exception e) {
                 return internalServerError("Während des Uploads ist etwas schiefgegangen!");
             }
-            return created("/media/"+med.id);
+
+            Folder rootFolder  = FolderManager.findRoot(folder);
+            List<Media> mediaSet = folder.files;
+            List<Folder> folderList = folder.folders;
+
+            return created(list.render(mediaSet, folderList, rootFolder.id));
```

Comment:
ah, and this one also breaks the file upload for posts
i guess its depricated since you have the mediaList call?